### DISCUSSION TITLE
Can GCN and GIN run in minibatch training like GAT does?

### DISCUSSION THREAD

**Discussion by imethanlee** [pyg-team/pytorch_geometric#4974] [Q&A] [answered]
I found an implementation of minibatch GAT (https://github.com/pyg-team/pytorch_geometric/blob/master/examples/ogbn_products_gat.py). I modify the code to GCN and GIN and it seems does not work. I was wondering whether it is possible to run GCN and GIN with minibatching as that GAT code does?

**Comment by rusty1s** [comment, marked as answer]
Yes, in general, all our GNN layers work in mini-batch mode. This was made easier with the new `NeighborLoader` class, which does no longer require bipartite message passing. [Here](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/reddit.py) is an example.
